analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsPattern.cs

```diff
@@ -32,26 +32,37 @@ internal sealed class IsPattern : BranchingProcessor<IIsPatternOperationWrapper>
 
     protected override ProgramState LearnBranchingConstraint(ProgramState state, IIsPatternOperationWrapper operation, bool falseBranch) =>
         operation.Value.TrackedSymbol(state) is { } testedSymbol
-        && LearnBranchingConstraint(state, operation.Pattern, falseBranch, state[testedSymbol]?.HasConstraint<ObjectConstraint>() is true) is { } constraint
+        && LearnBranchingConstraint(
+            state,
+            operation.Pattern,
+            falseBranch,
+            state[testedSymbol]?.HasConstraint<ObjectConstraint>() is true,
+            state.Constraint<NumberConstraint>(testedSymbol))
```

Comment:
can't this use the same syntax as above? `...HasConstraint() is true` style for consistency?

Reply:
In one case I want to know, whether or not the constraint is present. In the other case I need the actual constraint (I want to know the numbers).

Reply:
🤦 I read the brackets wrong, I thought the `is constraint` was inside `LearnBranchingConstraint`